core/node/dns.go

```diff
@@ -0,0 +1,27 @@
+package node
+
+import (
+	"fmt"
+	"strings"
+
+	config "github.com/ipfs/go-ipfs-config"
+	doh "github.com/libp2p/go-doh-resolver"
+	madns "github.com/multiformats/go-multiaddr-dns"
+)
+
+func DNSResolver(cfg *config.Config) (*madns.Resolver, error) {
```

Comment:
Sidenote regarding backward-compatibility: 
I believe we should include DoH for ENS as an implicit entry here.

- We want to remove hardcoded resolver hack from https://github.com/ipfs/go-ipfs/pull/6448  and https://github.com/ipfs/go-namesys/pull/8
- At the same time, we can't break resolution of `.eth` names in default config at this point.
  - We also don't want to hardcode it in https://github.com/ipfs/go-ipfs-config/ as that is noisy, and makes it hard to migrate default like this in the future. 
  - User should be able to override the default (implicit) ENS resolver  by adding an explicit `eth.` key to the config,

I believe the least painful way is to `append` an implicit DoH resolver ( `eth.` → `https://eth.link/dns-query`) to the list if the config is missing an explicit key for `eth.`

Reply:
> I believe the least painful way is to append an implicit DoH resolver ( eth. → https://eth.link/dns-query) to the list if the config is missing an explicit key for eth

Do you mean add it into the config file (i.e. via a migration), or add it in the code that creates the resolver?

If via the config file, what if the user just wants to use a single resolver for `.` and so removes `eth.`, will we add it back in?

If we want `ipfs init` to always result in a config + binary that has a DoH binding specifically for `eth.` (as we do now) then we're going to need to some code somewhere with that mapping.

Reply:
Do we have to do this in this pr? Sounds like we might want to do in a subsequent one.

Reply:
Note that I am fine with adding an implicit entry for `eth.`.

Reply:
updated.

Reply:
@aschmahmann I meant code that creates resolver. We don't want to pollute configs with brittle default that can change over  time.  
@vyzo implemented it the way I imagined.